src/OSPSuite.UI/Binders/CurveBinder.cs

```diff
@@ -356,6 +355,19 @@ private void refreshData()
          _dataTable.EndLoadData();
       }
 
+      /// <summary>
+      /// Compares the <paramref name="dataColumn"/> base grid value at the <paramref name="baseIndex"/>. If it
+      /// is the same as <paramref name="baseValue"/> then we can use indexing to retrieve the value, otherwise use interpolation
+      /// </summary>
+      /// <returns>The value of the data column in base units for the given <paramref name="baseValue"/></returns>
+      protected static float ValueInBaseUnit(DataColumn dataColumn, float baseValue, int baseIndex)
+      {
+         if ((dataColumn.BaseGrid.Count > baseIndex) && (Math.Abs(dataColumn.BaseGrid[baseIndex] - baseValue) < float.Epsilon))
```

Comment:
is there an extension method for float epsilon checks?

Reply:
No idea

Reply:
What is this check doing.? This looks weird. So if i understand is correctly, you are checking that the value at the index is similar to the one that I have in the column basegrid. If yes , then we assume we can use index?

Does it make more sense to check if the basegrid are the same ? If yes, then you can return index. If not , then extrapolation?

Reply:
Yeah, we could do that. What is the prevalence of cases where the base grid has the same values but is not the same object? We would lose the optimization in those cases.
Conversely just refactor a ```hasSameBaseValueAt(int baseIndex)``` method to make it more easy to understand.

Reply:
What if we used the same for related columns? Would those have the same basegrid, or a different basegrid with the same values?

```AddRelatedValuesToRow```

Reply:
So do we need to worry about a CurveBinder having references to different base grids?

Reply:
Yeah, it's a bit confusing both x and y data have reference to their own basegrid, which is probably the same one.

```
      private BaseGrid activeBaseGrid(DataColumn xData, DataColumn yData)
      {
         if (Curve.InterpolationMode == InterpolationModes.xLinear)
            return xData.BaseGrid;

         if (Curve.InterpolationMode == InterpolationModes.yLinear)
            return yData.BaseGrid;

         throw new ArgumentException("InterpolationMode = " + Curve.InterpolationMode);
      }
```

Anyway, I think we're down to checking if the base grids match, and if so, we return the column data value for that index. If the base grids don't reference match, revert to the previous strategy.